packages/core/src/hooks/useSlowTransactions.ts

```diff
@@ -0,0 +1,70 @@
+import { useBlockNumber, useConfig } from '../providers'
+import { useEthers } from './useEthers'
+import React, { useEffect, useState } from 'react'
+
+import { Web3Provider } from '@ethersproject/providers'
+
+async function updateTransactions(
+  setTransactions: React.Dispatch<React.SetStateAction<{ hash: string; submittedAt: number }[]>>,
+  setSlowTransactions: React.Dispatch<React.SetStateAction<{ hash: string; submittedAt: number }[]>>,
+  transactions: { hash: string; submittedAt: number }[],
+  threshhold: number,
+  library: Web3Provider
+) {
+  const results = await Promise.all(
+    transactions.map(async ({ hash }) => {
+      return await library?.getTransaction(hash)
+    })
+  )
```

Comment:
Requesting transactions one by one will probably be safer. Right now it can start nearly an unlimited amount of web requests at once.  
`map` is not working well with async code so it's not waiting for the request to finish before starting the next one.